crates/hyperdrive-math/src/long/max.rs

```diff
@@ -574,7 +574,17 @@ mod tests {
                         FixedPoint::from(expected_base_amount)
                     );
                 }
-                Err(_) => assert!(actual.is_err() || actual.unwrap().is_err()),
+                Err(e) => {
+                    if !actual.is_err() {
+                        let unwrapped = actual.unwrap();
```

Comment:
is this right? My understanding is that unwrap does implicit error handling and will either return the inner element or panic. So you'll never get to the assert if it is an error.